addon/mixins/route.js

```diff
@@ -160,35 +172,24 @@ export default Ember.Mixin.create({
     delete options.perPage;
     delete options.modelPath;
 
-    this.set('_perPage', perPage);
-    this.set('_modelPath', modelPath);
-    this.set('_extraParams', options);
-
-    var requestPayloadBase = {};
-    requestPayloadBase[this.get('perPageParam')] = perPage;
-    requestPayloadBase[this.get('pageParam')] = startingPage;
```

Comment:
The problem before was that the initial/default value of `currentPage` wasn't used; instead `startingPage` was initialized like `var startingPage = options.startingPage || 1;` and then later on `this.set('currentPage', startingPage);`